### PR TITLE
build: separate stdlib manifests by version

### PR DESCRIPTION
**Pull request by vtjnash**
We install these by version, so it helps a lot if they reflect that in the manifest name.

### CODE REVIEW

File: stdlib/Makefile [hunk 1]
```diff
@@ -43,14 +42,19 @@ $(foreach jll,$(JLLS),$(eval $(call download-artifacts-toml,$(jll))))
...
-          SharedArrays Sockets SparseArrays SuiteSparse Test TOML Unicode UUIDs \
+          Markdown Mmap Printf Profile Random REPL Serialization \
+          SharedArrays Sockets Test TOML Unicode UUIDs \
           $(JLL_NAMES)
 
 STDLIBS_EXT = Pkg Statistics LibCURL Downloads ArgTools Tar NetworkOptions SuiteSparse SparseArrays SHA
 
 $(foreach module, $(STDLIBS_EXT), $(eval $(call stdlib-external,$(module),$(shell echo $(module) | tr a-z A-Z))))
 
+ifneq ($(filter $(STDLIBS),$(STDLIBS_EXT)),)
```
Comment (staticfloat) [inline stdlib/Makefile:53]: hah

**Review by staticfloat — APPROVED** [review]

File: Make.inc [hunk 2]
```diff
@@ -1412,6 +1412,8 @@ endif
 define dir_target
 $$(abspath $(1)):
 	@mkdir -p $$@
+$$(abspath $(1))/: | $$(abspath $(1))
+	@true
```
Comment (vtjnash) [inline Make.inc:1416]: Don't merge yet. apparently the old version of make on mac used to understand that these were the same path, but the updated versions used everywhere else aren't able to handle this anymore. I guess we need to add this trailing `/` everywhere instead of making it convenient with this added rule

File: Makefile [hunk 3]
```diff
@@ -44,10 +61,6 @@ $(foreach link,base $(JULIAHOME)/test,$(eval $(call symlink_target,$(link),$$(bu
 julia_flisp.boot.inc.phony: julia-deps
 	@$(MAKE) $(QUIET_MAKE) -C $(BUILDROOT)/src julia_flisp.boot.inc.phony
 
-# Build the HTML docs (skipped if already exists, notably in tarballs)
-$(BUILDROOT)/doc/_build/html/en/index.html: $(shell find $(BUILDROOT)/base $(BUILDROOT)/doc \( -path $(BUILDROOT)/doc/_build -o -path $(BUILDROOT)/doc/deps -o -name *_constants.jl -o -name *_h.jl -o -name version_git.jl \) -prune -o -type f -print)
-	@$(MAKE) docs
```
Comment (nalimilan) [inline Makefile:49]: Why did you delete this? If I'm not mistaken, this was necessary to avoid rebuilding docs when building and installing from tarballs. And rebuilding docs downloads Unicode.txt, which requires Internet access (disabled on distribution build machines to ensure reproducibility), and makes the build dependent on the existence of the file at the expected URL.

Comment (vtjnash) [reply Makefile:49]: This was the wrong way to write makefile dependency rules. I think it should still build it, but respecting timestamps.

Comment (nalimilan) [reply Makefile:49]: Sorry, actually this commit isn't included in 1.9.0-beta4 and yet I have a problem where docs get rebuild there, so it's caused by something different.
